scio-core/src/main/scala/com/spotify/scio/values/SCollection.scala

```diff
@@ -131,18 +134,33 @@ sealed trait SCollection[T] extends PCollectionWrapper[T] {
   def setCoder(coder: org.apache.beam.sdk.coders.Coder[T]): SCollection[T] =
     context.wrap(internal.setCoder(coder))
 
+  def setSchema(schema: Schema[T]): SCollection[T] = {
+    if (!internal.hasSchema) {
+      val (s, to, from) = SchemaMaterializer.materialize(this.context, schema)
+      context.wrap(internal.setSchema(s, to, from))
+    } else this
+  }
+
+  private def checkCoderSerialization[T](coder: BCoder[T]): Unit =
```

Comment:
wdyt of returning `Either` here? I know that we have this style across the code base but I would like to start changing this behavior

Reply:
This is somewhat out of the scope of this PR. I'd prefer to make that change in another one

Reply:
+1